packages/@lwc/engine-core/src/framework/reporting.ts

```diff
@@ -43,7 +76,7 @@ export const reportingControl = {
      *
      * @param dispatcher - reporting control
      */
-    attachDispatcher(dispatcher: ReportingDispatcher): void {
+    attachDispatcher(dispatcher: ReportingDispatcher<ReportingEventId>): void {
```

Comment:
I'm sure there is some fancy TypeScript way to make this work:

```ts
reportingControl.attachDispatcher(<T extends ReportingEventId>(reportingEventId: T, payload: Payload<T>) => {
    if (reportingEventId === ReportingEventId.StylesheetMutation) {
        // TypeScript should know this is a StylesheetMutationPayload, but it doesn't
        console.log(payload.propertyName)
    } else if (reportingEventId === ReportingEventId.CrossRootAriaInSyntheticShadow) {
        // TypeScript should know this is a CrossRootAriaPayload, but it doesn't
        console.log(payload.attributeName)
    }
})
```

... but I couldn't figure it out. It's not the end of the world, though – just a bit of extra typing required by consumers of this API.